src/textual/widgets/_loading_indicator.py

```diff
@@ -22,8 +24,49 @@ class LoadingIndicator(Widget):
         content-align: center middle;
         color: $accent;
     }
+    LoadingIndicator.-overlay {
+        overlay: screen;
+        background: $boost;
+    }
     """
 
+    def apply(self, widget: Widget) -> AwaitMount:
+        """Apply the loading indicator to a `widget`.
+
+        This will overlay the given widget with a loading indicator.
+
+        Args:
+            widget: A widget.
+
+        Returns:
+            AwaitMount: An awaitable for mounting the indicator.
+        """
+        self.add_class("-overlay")
+        await_mount = widget.mount(self, before=0)
+        return await_mount
+
+    @classmethod
+    def clear(cls, widget: Widget) -> Awaitable:
```

Comment:
Just to note, and perhaps it's something to handle or not (or at least document): turns out that `get_child_by_type` is really `get_first_child_by_type`; so if by some bizarre situation there are multiple loading indicators here, it'll remove the first one it finds (which is fine).

I do wonder though if this should perhaps remove *all* the loading indicators it finds?